modules/core/include/opencv2/core/cuda.hpp

```diff
@@ -656,6 +656,18 @@ class CV_EXPORTS_W Stream
     //! creates a new asynchronous stream with custom allocator
     CV_WRAP Stream(const Ptr<GpuMat::Allocator>& allocator);
 
+    /** @brief creates a new Stream using the cudaFlags argument to determine the behaviors of the stream
+
+    @note The cudaFlags parameter is passed unchanged to the underlying api cudaStreamCreateWithFlags() and
+    supports the same parameter values.
+    @code
+        // creates an OpenCV cuda::Stream that manages an asynchronous, non-blocking,
+        // non-default CUDA stream
+        cv::cuda::Stream cvStream(cudaStreamNonBlocking);
+    @endcode
+     */
+    CV_WRAP Stream(const unsigned cudaFlags);
```

Comment:
Remove CV_WRAP from here to keep this C++ only.

We can add `CV_WRAP static Stream create(int cudaFlags)` or similar for bindings later.

Reply:
I thought of an alternative. I will need to code it and force push to see if it removes the two warnings in the CI platforms. (Functionality has not been a problem...only two warnings on a subset of platforms).

I can code the constructor to accept an `int` (you suggest this is preferred in OpenCV) and also reject negative numbers. By accepting only positive `int` values...I can safely cast it to an `unsigned int`. This alternative supports all the values that the NVidia API currently supports. If in the future NVidia defines a value that is larger than a signed int can represent and someone uses that value, then this constructor will fail. And OpenCV will need to make a choice how to handle NVidia's new value.

I would prefer to not have the cost of a branch/if in the constructor. What is the OpenCV philosophy to use `CV_DbgAssert(param>=0)`. I would prefer that so there is no cost in release builds.
If the core team doesn't like that, then I can use `CV_Assert(param>=0)`.